### PR TITLE
ER-1278-Add new GTM snippets

### PR DESCRIPTION
**Pull request by shomavg**

### CODE REVIEW

File: src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs [hunk 1]
```diff
@@ -13,11 +13,9 @@ public static class EnvironmentNames
         public const string PROD = "PROD";
 
         public static string[] GetTestEnvironmentNames() => new []{ Development, AT, TEST, DEMO };
-        
         public static string GetTestEnvironmentNamesCommaDelimited() => string.Join(",", Development, AT, TEST, DEMO);
-
+        public static string GetEnvironmentNamesCommaDelimitedForGMT() => string.Join(",", DEMO, PREPROD, PROD);
```
Comment (mgwilliam) [inline src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: should be GTM not GMT :o)

File: src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs [hunk 2]
```diff
@@ -13,11 +13,9 @@ public static class EnvironmentNames
         public const string PROD = "PROD";
```
Comment (mgwilliam) [inline src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: TEST2 should be a test environment too

**Comment by shomavg** [comment]
> This is excluding DEMO, PREPROD & PROD. Is this right?

-Notes from Mark:
Must be emitted for DEMO, PRE-PROD and PROD environments.
we include the DEMO environment

note - we exclude for AT/TEST/TEST2

**Comment by chrisjensenuk** [comment]
<environment *exclude*="@EnvironmentNames.GetEnvironmentNamesCommaDelimitedForGTM()">

File: src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs [hunk 3]
```diff
@@ -8,13 +8,14 @@ public static class EnvironmentNames
...
         public const string TEST = "TEST";
+        public const string TEST2 = "TEST2";
         public const string DEMO = "DEMO";
         public const string PREPROD = "PREPROD";
         public const string PROD = "PROD";
 
         public static string[] GetTestEnvironmentNames() => new []{ Development, AT, TEST, DEMO };
         public static string GetTestEnvironmentNamesCommaDelimited() => string.Join(",", Development, AT, TEST, DEMO);
-        public static string GetEnvironmentNamesCommaDelimitedForGMT() => string.Join(",", DEMO, PREPROD, PROD);
+        public static string GetEnvironmentNamesCommaDelimitedForGTM() => string.Join(",", DEMO, PREPROD, PROD);
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: GetTestEnvironmentNamesCommaDelimited & GetNonProdEnvironmentNamesCommaDelimited need to include TEST2. Maybe more?

File: src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs [hunk 4]
```diff
@@ -8,16 +8,14 @@ public static class EnvironmentNames
         public const string Development = "DEVELOPMENT";
         public const string AT = "AT";
         public const string TEST = "TEST";
+        public const string TEST2 = "TEST2";
         public const string DEMO = "DEMO";
         public const string PREPROD = "PREPROD";
         public const string PROD = "PROD";
 
         public static string[] GetTestEnvironmentNames() => new []{ Development, AT, TEST, DEMO };
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: Add TEST2

File: src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs [hunk 5]
```diff
@@ -8,16 +8,14 @@ public static class EnvironmentNames
...
-        
-        public static string GetTestEnvironmentNamesCommaDelimited() => string.Join(",", Development, AT, TEST, DEMO);
-
+        public static string GetTestEnvironmentNamesCommaDelimited() => string.Join(",", Development, AT, TEST, TEST2);
         public static string GetNonProdEnvironmentNamesCommaDelimited() => string.Join(",", Development, AT, TEST, DEMO, PREPROD);
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: Add TEST2

File: src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs [hunk 6]
```diff
@@ -8,16 +8,14 @@ public static class EnvironmentNames
...
         public const string PREPROD = "PREPROD";
         public const string PROD = "PROD";
 
-        public static string[] GetTestEnvironmentNames() => new []{ Development, AT, TEST, DEMO };
-        
-        public static string GetTestEnvironmentNamesCommaDelimited() => string.Join(",", Development, AT, TEST, DEMO);
-
-        public static string GetNonProdEnvironmentNamesCommaDelimited() => string.Join(",", Development, AT, TEST, DEMO, PREPROD);
-
+        public static string[] GetTestEnvironmentNames() => new []{ Development, AT, TEST, TEST2, DEMO };
```
Comment (sachpatel) [inline src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: For complicated reason this method conflicts with the one below regarding what are "test" envs. One includes DEMO and the other doesn't and they want the GTM snippet to be output for the DEMO env. Suggestion is to remove DEMO from the `GetTestEnvironmentNames` method.

Comment (chrisjensenuk) [reply src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: Could `GetTestEnvironmentNamesCommaDelimited` be created from` GetTestEnvironmentNames`?

Comment (shomavg) [reply src/Shared/Recruit.Shared.Web/Configuration/EnvironmentNames.cs]: The reason I dint use GetTestEnvironmentNames() is because it had Demo in it.

**Pull request merged**